lib/jsonld-signatures.js

```diff
@@ -364,6 +357,32 @@ function _getSupportedAlgorithms() {
   return Object.keys(api.suites).filter(s => s !== 'LinkedDataSignature');
 }
 
+function _isSupportedProofPurpose(purpose) {
+  if(!purpose) {
+    // TODO: Enable error in the next major release
+    // throw new Error(`Please provide 'purpose' of the proof to verify.`);
+    return;
+  }
+  try {
+    api.proofPurposes.use(purpose);
+  } catch(e) {
+    if(e.message === `Cannot find module '${purpose}'`) {
+      throw new Error(`Unsupported proof purpose handler "${purpose}".`);
+    }
+    throw e;
+  }
+}
+
+async function _getExpandedProofPurpose({proof, jsonldOpts}) {
+  const {proofPurpose} = proof;
+  const jsonld = injector.use('jsonld');
+
+  const expanded = await jsonld.expand({
+    proofPurpose,
+    '@context': constants.SECURITY_CONTEXT_URL}, jsonldOpts);
+  return expanded[0]['https://w3id.org/security#proofPurpose'][0]['@id'];
```

Comment:
Everything has been validated up to this point to ensure that this is not going to fail, correct?

Reply:
If by validation, you mean code to check that this structure exists... the answer is no. I'm assuming that this structure is guaranteed by the use of the security context and the term definition of `proofPurpose` in that context. May need @dlongley @msporny @davidlehn to verify my JSON-LD logic.

Reply:
What if `proofPurpose` is undefined? What if `proofPurpose` does not map to `https://w3id.org/security#proofPurpose`? If there aren't other logical guarantees to ensure these things are true before the above code is run, it could blow up. We should have tests.